app/services/prisoner_location.rb

```diff
@@ -0,0 +1,46 @@
+class PrisonerLocation
+  INVALID = 'location_invalid'.freeze
+  UNKNOWN = 'location_unknown'.freeze
+
+  include ActiveModel::Model
+
+  validate :located_at_given_prison
+  validate :validate_has_location
+
+  def initialize(offender, prison_code = nil)
+    self.offender    = offender
+    self.prison_code = prison_code
+  end
+
+private
+
+  attr_accessor :offender, :prison_code
+
+  def validate_has_location
+    unless prisoner_location.api_call_successful?
+      errors.clear
```

Comment:
Why do you need to clear the errors? Usually errors are clear at the beginning of the validation lifecycle

Reply:
That's to ensure that if an API error happens we only get this objects. 
Both validations get run and we'd get two errors in the case of an API error, doesn't really makes sense to have an `invalid location` when the API actually failed.
Another way to avoid `errors.clear` was to run `validate_has_location` first and then but `located_at_given_prison`
and check for `api_call_successful?` but I didn't quite like it because it implicitely requires validations to be run in a specific order.